interp/src/values.rs

```diff
@@ -390,6 +397,11 @@ pub struct PulseValue {
 }
 
 impl PulseValue {
+    ///(Assuming high_val, low_val1, and low_val2 must all have the same width)
+    ///Returns the width of this value
```

Comment:
Yes they should! Good point. Could you add an assert statement in the constructor for PulseValue?